### PR TITLE
Add missing type annotations

### PR DESCRIPTION
**Pull request by colerogers**
### Description

Enhance CORS options by adding RegEx, RegEx arrays, and string arrays.

### Code sample

`cors?: string | boolean | RegExp | (string | RegExp)[];`

### CODE REVIEW

**Comment by colerogers** [comment]
I didn't add any additional tests since we'd be essentially testing the cors library. Let me know if any of you disagree and I can add some

**Comment by inlined** [comment]
lmfao. Was that really all the work that was required?

**Review by inlined — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- Enhance CORS config options
```
Comment (taeold) [inline CHANGELOG.md:1]: Do we want v2 changes to be included in the changelog?

Comment (colerogers) [reply CHANGELOG.md:1]: @inlined Suggested adding a changelog to this, but it makes sense not to have v2 changes in there

**Comment by shauryaaher** [comment]
Guys, looks good, merge this PR.

**Pull request merged**